src/Paprika/RLP/KeccakOrRlp.cs

```diff
@@ -0,0 +1,48 @@
+using Paprika.Crypto;
+
+namespace Paprika.RLP;
+
+public readonly ref struct KeccakOrRlp
```

Comment:
I also tried a different approach that uses a fixed size array instead of a `Keccak`:

```c#
public ref struct KeccakOrRlp
{
    public enum Type : byte
    {
        Keccak = 0,
        Rlp = 1,
    }

    private unsafe ref struct Buffer
    {
        private const int Size = 32;

#pragma warning disable CS0649
        // Compiler cannot figure out that the fields is being used
        private fixed byte _data[Size];
#pragma warning restore CS0649

        public Span<byte> AsSpan()
        {
            fixed (byte* p = _data)
            {
                return new Span<byte>(p, Size);
            }
        }
    }

    public Type DataType { get; }
    private Buffer _buffer;

    public Span<byte> AsSpan() => _buffer.AsSpan();

    private KeccakOrRlp(Type dataType, Buffer buffer)
    {
        DataType = dataType;
        _buffer = buffer;
    }

    public static KeccakOrRlp FromSpan(scoped Span<byte> data)
    {
        var buffer = new Buffer();
        var output = buffer.AsSpan();

        if (data.Length < 32)
        {
            output[0] = (byte)data.Length;
            data.CopyTo(output[1..]);
            return new KeccakOrRlp(Type.Rlp, buffer);
        }
        else
        {
            KeccakHash.ComputeHash(data, output);
            return new KeccakOrRlp(Type.Keccak, buffer);
        }
    }
}
```

On benchmarks, this approach is as performant as the `Keccak` implementation, while in my opinion being a bit clearer since we're not using a `Keccak` as a backing field (which, when we store an `Rlp` is not even a `Keccak`).

I'm not fully sure though if we're performing additional copies though, but I don't see any on the IL (besides the required ones).

Reply:
This should work as you describe.

Reply:
Should we use this approach then? Or would you rather stick with the Keccak approach?